app/models/clubhouse.rb

```diff
@@ -0,0 +1,36 @@
+class Clubhouse < ApplicationRecord
+  has_many :posts, dependent: :destroy
+  has_many :memberships, dependent: :destroy
+
+  validates :title, presence: true
+
+  def self.with_owner(owner, properties)
+    clubhouse = Clubhouse.create(properties)
+    membership = Membership.new(
+      user: owner,
+      clubhouse:,
+      role: 'owner'
+    )
+    return clubhouse if membership.save
+
+    nil
+  end
+
+  def owners
+    users_of_role('owner')
+  end
+
+  def admins
+    users_of_role('admin')
+  end
+
+  def members
+    users_of_role('member')
+  end
```

Comment:
You can call the constants I suggested you to create on the membership model to avoid having magic strings here around
```suggestion
  def owners
    users_of_role(Membership::OWNER_ROLE)
  end

  def admins
    users_of_role(Membership::ADMIN_ROLE)
  end

  def members
    users_of_role(Membership::MEMBER_ROLE)
  end
```

Reply:
Done!